tests/std/tests/P0896R4_ranges_alg_copy_backward/test.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate that copy_backward_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::copy_backward_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::copy_backward(borrowed<true>{}, static_cast<int*>(nullptr))),
+    ranges::copy_backward_result<int*, int*>>);
+
+static constexpr int input[] = {13, 42, 1729};
+
+struct instantiator {
+    template <class Bidi1, class Bidi2>
+    static constexpr void call() {
+        // For the second range, we need an iterator to the end; it's expedient to simply ignore ranges with differing
+        // iterator and sentinel types (i.e., ranges that don't satisfy common_range).
+        if constexpr (ranges::common_range<Bidi2>) {
+            using ranges::copy_backward, ranges::copy_backward_result, ranges::equal, ranges::iterator_t;
+            const Bidi1 wrapped_input{input};
+
+            { // Validate range overload
+                int output[] = {-2, -2, -2};
+                Bidi2 wrapped_output{output};
+                auto result = copy_backward(wrapped_input, wrapped_output.end());
+                STATIC_ASSERT(same_as<decltype(result), copy_backward_result<iterator_t<Bidi1>, iterator_t<Bidi2>>>);
+                assert(result.in == wrapped_input.end());
+                assert(result.out == wrapped_output.begin());
+                assert(equal(output, input));
+            }
+            { // Validate iterator + sentinel overload
+                int output[] = {-2, -2, -2};
+                Bidi2 wrapped_output{output};
+                auto result = copy_backward(wrapped_input.begin(), wrapped_input.end(), wrapped_output.end());
+                STATIC_ASSERT(same_as<decltype(result), copy_backward_result<iterator_t<Bidi1>, iterator_t<Bidi2>>>);
+                assert(result.in == wrapped_input.end());
+                assert(result.out == wrapped_output.begin());
+                assert(equal(output, input));
+            }
+        }
+    }
+};
+
+constexpr void test_memmove() {
+    // Get some coverage for the memmove optimization, which we would not otherwise have since we do not (as of
+    // 20200709) unwrap output iterators.

```

Comment:
I'd recommend 2020-07-09 as easier to read. I'd also recommend directly citing GH-893, especially because the GitHub Issues and Pull Requests extension for VSCode will recognize such citations in its next update.